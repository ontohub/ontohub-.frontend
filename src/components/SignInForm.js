import React, { Component } from 'react'
import { Button, Form, Message } from 'semantic-ui-react'

export class SignInForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      error: false
    }
    this.onSubmit = this.onSubmit.bind(this)
  }
  onSubmit(event) {
    event.preventDefault()
    return this.props.onSubmit(this.username.value, this.password.value).then(
      () => {
        this.setState({ error: false })
        this.props.onSuccess()
      },
      () => {
        this.setState({ error: true })
        this.props.onError()
      }
    )
  }
  render() {
    return (
      <Form error={this.state.error} onSubmit={this.onSubmit}>
        {this.state.error &&
          <Message
            error
            icon="warning sign"
            header="Incorrect username or password"
          />}
        <Form.Group widths="equal">
          <Form.Field error={this.state.error}>
            <label htmlFor="sign-in-username">Username</label>
            <input
              defaultValue="ada"
              ref={(input) => (this.username = input)}
              placeholder="Username"
              id="sign-in-username"
            />
          </Form.Field>
          <Form.Field error={this.state.error}>
            <label htmlFor="sign-in-password">Password</label>
            <input
              type="password"
              defaultValue="changemenow"
              ref={(input) => (this.password = input)}
              placeholder="Password"
              id="sign-in-password"
            />
          </Form.Field>
        </Form.Group>
        <Button
          type="submit"
          icon="sign in"
          content="Sign in"
          floated="right"
          primary
        />
      </Form>
    )
  }
}

export default SignInForm
