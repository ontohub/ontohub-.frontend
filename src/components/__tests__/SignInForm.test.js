import React from "react";
import { shallow, mount } from "enzyme";
import SignInForm from "../SignInForm";

describe("SignInForm", () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<SignInForm />);
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("onSubmit", () => {
    describe("error", () => {
      let onError;

      beforeAll(() => {
        onError = jest.fn();
        wrapper = mount(
          <SignInForm onSubmit={() => Promise.reject()} onError={onError} />
        );
        wrapper
          .instance()
          .onSubmit({ preventDefault: jest.fn() })
          // Somehow enzyme does not update the wrapper's state
          .then(() => wrapper.setState({ error: true }));
      });

      it("matches the snapshot", () => {
        expect(wrapper).toMatchSnapshot();
      });

      it("sets the error state to true", () => {
        expect(wrapper.state("error")).toBeTruthy();
      });
      it("calls the error callback", () => {
        expect(onError.mock.calls.length).toBe(1);
      });
    });

    describe("success", () => {
      let onSuccess;

      beforeAll(() => {
        onSuccess = jest.fn();
        wrapper = mount(
          <SignInForm
            onSubmit={() => Promise.resolve()}
            onSuccess={onSuccess}
          />
        );
        wrapper.instance().onSubmit({ preventDefault: jest.fn() });
      });

      it("sets the error state to false", () => {
        expect(wrapper.state("error")).toBeFalsy();
      });
      it("calls the success callback", () => {
        expect(onSuccess.mock.calls.length).toBe(1);
      });
    });
  });
});
