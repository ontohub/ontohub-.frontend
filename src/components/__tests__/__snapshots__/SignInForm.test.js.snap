// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm matches the snapshot 1`] = `
<Form
  as="form"
  error={false}
  onSubmit={[Function]}
>
  <FormGroup
    widths="equal"
  >
    <FormField
      error={false}
    >
      <label
        htmlFor="sign-in-username"
      >
        Username
      </label>
      <input
        defaultValue="ada"
        id="sign-in-username"
        placeholder="Username"
      />
    </FormField>
    <FormField
      error={false}
    >
      <label
        htmlFor="sign-in-password"
      >
        Password
      </label>
      <input
        defaultValue="changemenow"
        id="sign-in-password"
        placeholder="Password"
        type="password"
      />
    </FormField>
  </FormGroup>
  <Button
    as="button"
    content="Sign in"
    floated="right"
    icon="sign in"
    id="sign-in-form-sign-in-button"
    primary={true}
    type="submit"
  />
</Form>
`;

exports[`SignInForm onSubmit error matches the snapshot 1`] = `
<SignInForm
  onError={[Function]}
  onSubmit={[Function]}
>
  <Form
    as="form"
    error={true}
    onSubmit={[Function]}
  >
    <form
      className="ui error form"
      onSubmit={[Function]}
    >
      <Message
        error={true}
        header="Incorrect username or password"
        icon="warning sign"
      >
        <div
          className="ui error icon message"
        >
          <Icon
            as="i"
            name="warning sign"
          >
            <i
              aria-hidden="true"
              className="warning sign icon"
            />
          </Icon>
          <MessageContent>
            <div
              className="content"
            >
              <MessageHeader
                content="Incorrect username or password"
              >
                <div
                  className="header"
                >
                  Incorrect username or password
                </div>
              </MessageHeader>
            </div>
          </MessageContent>
        </div>
      </Message>
      <FormGroup
        widths="equal"
      >
        <div
          className="equal width fields"
        >
          <FormField
            error={true}
          >
            <div
              className="error field"
            >
              <label
                htmlFor="sign-in-username"
              >
                Username
              </label>
              <input
                defaultValue="ada"
                id="sign-in-username"
                placeholder="Username"
              />
            </div>
          </FormField>
          <FormField
            error={true}
          >
            <div
              className="error field"
            >
              <label
                htmlFor="sign-in-password"
              >
                Password
              </label>
              <input
                defaultValue="changemenow"
                id="sign-in-password"
                placeholder="Password"
                type="password"
              />
            </div>
          </FormField>
        </div>
      </FormGroup>
      <Button
        as="button"
        content="Sign in"
        floated="right"
        icon="sign in"
        id="sign-in-form-sign-in-button"
        primary={true}
        type="submit"
      >
        <button
          className="ui primary right floated button"
          id="sign-in-form-sign-in-button"
          onClick={[Function]}
          type="submit"
        >
          <Icon
            as="i"
            name="sign in"
          >
            <i
              aria-hidden="true"
              className="sign in icon"
            />
          </Icon>
          Sign in
        </button>
      </Button>
    </form>
  </Form>
</SignInForm>
`;
